lxd/main_netcat.go

```diff
@@ -17,31 +19,55 @@ import (
 // after the path to the unix socket are ignored, so that this can be passed
 // directly to rsync as the sync command.
 func cmdNetcat(args []string) error {
-	if len(args) < 2 {
+	if len(args) < 3 {
 		return fmt.Errorf("Bad arguments %q", args)
 	}
 
+	logPath := shared.LogPath(args[2], "netcat.log")
+	if shared.PathExists(logPath) {
+		os.Remove(logPath)
+	}
+
+	logFile, err := os.OpenFile(logPath, os.O_WRONLY|os.O_CREATE|os.O_SYNC, 0644)
+	if err != nil {
+		logFile = nil
+	} else {
+		defer logFile.Close()
+	}
```

Comment:
Lets just make logfile creation failure fatal.
If /var/log/lxd/ can't be written to, you have much bigger problems than a failed container migration :)

That would also simplify the rest of this change by not having to constantly check for logFile != nil